File: collatz.py
import matplotlib.pyplot as plt
import networkx as nx

def is_even(number):
    if number%2 == 0:
        return True
    else:
        return False

series = []
xaxis = []
daglist = []
def collatz(number):
    series.append(int(number))
    if is_even(number):
        number = number/2
        #series.append(int(number))
        collatz(number)
    elif not is_even(number) and number != 1:
        number = 3*number + 1
        #series.append(int(number))
        collatz(number)
    elif number == 1:
        pass

def print_collatz(number):
    collatz(number)
    print(series)

def graph(number):
    collatz(number)
    for i in range(len(series)):
        xaxis.append(i)
    plt.plot(xaxis, series)
    #plt.show()

for number in range(1, 10001):
    print_collatz(number)
    #collatz(number)
    for i in range(len(series)):
        xaxis.append(i)
    plt.plot(xaxis, series)
    series = []
    xaxis =[]

plt.show()
#print(daglist)

File: Collatz.java
import java.util.*;
import java.util.function.Supplier;

public class Collatz{
  static List<Integer> series = new ArrayList<Integer>();
  public static int collatz(int number){
    series.add(number);
    if(number != 1){
      if(number%2 == 0){
        number = number/2;
        collatz(number);
      }else if(number%2 == 1){
        number = 3*number + 1;
        collatz(number);
      }
    }
  return 0;
  }

  public static void main(String[] args){
    int number = 10000;
    for(int i = 1; i < number; i++){
      collatz(i);
      System.out.println(series.toString());
      series.removeAll(series);
    }
  }
}
